src/UCommerce.SiteCore.Installer/Steps/MoveUcommerceBinaries.cs

```diff
@@ -1,103 +1,110 @@
 ﻿using System.Collections.Generic;
-using System.Collections.Specialized;
-using Sitecore.Install.Framework;
+using System.IO;
+using System.Threading.Tasks;
+using Ucommerce.Installer;
 
 namespace Ucommerce.Sitecore.Installer.Steps
 {
-    public class MoveUcommerceBinaries : IPostStep
+    public class MoveUcommerceBinaries : IStep
     {
-        public void Run(ITaskOutput output, NameValueCollection metaData)
+        private readonly DirectoryInfo _basePath;
+        private readonly IInstallerLoggingService _logging;
+        private readonly DirectoryInfo _sitecoreBasePath;
+
+        public MoveUcommerceBinaries(DirectoryInfo basePath, DirectoryInfo sitecoreBasePath, IInstallerLoggingService logging)
+        {
+            _basePath = basePath;
+            _sitecoreBasePath = sitecoreBasePath;
+            _logging = logging;
+        }
+
+        public async Task Run()
         {
-            var _postInstallationSteps = new List<IPostStep>();
+            _logging.Information<MoveUcommerceBinaries>("Moving Ucommerce binaries");
+            var installationSteps = new List<IStep>();
 
             // Move the Commerce Connect dependend files to the Commerce Connect app location.
             // Therefor any older version located in the bin folder needs to be removed.
             //_postInstallationSteps.Add(new DeleteFile("~/bin/Ucommerce.Sitecore.CommerceConnect.dll"));
 
-            _postInstallationSteps.Add(new MoveFile(
-                "~/bin/ucommerce/Ucommerce.Infrastructure.dll",
-                "~/bin/Ucommerce.Infrastructure.dll",
+            installationSteps.Add(new MoveFile(new FileInfo(Path.Combine(_basePath.FullName, "bin", "ucommerce", "Ucommerce.Infrastructure.dll")),
+                new FileInfo(Path.Combine(_sitecoreBasePath.FullName, "bin", "Ucommerce.Infrastructure.dll")),
                 backupTarget: false));

```

Comment:
Doesnt the new MoveFile take 4 arguments? One more for the logger? I might be mistaken

Reply:
You're wrong :)